examples/crowdfunding/assets/crowdFund.teal

```diff
@@ -0,0 +1,469 @@
+byte "EndDate"
+txna ApplicationArgs 1
+btoi
+app_global_put
+
+// store fund goal
+byte "Goal"
+txna ApplicationArgs 2
+btoi
+app_global_put
+
+// store the fund receiver
+byte "Receiver"
+txna ApplicationArgs 3
+app_global_put
+
+// set the total raised to 0 and store it
+byte "Total"
+int 0
+app_global_put
+
+// store the fund close date
+byte "FundCloseDate"
+txna ApplicationArgs 4
+btoi
+app_global_put
+
+// return a success
+int 1
+return
+
+not_creation:
+// check if this is deletion transaction
+int DeleteApplication
+txn OnCompletion
+==
+bz not_deletion
+
+// To delete the app
+// The creator must empty the escrow
+// to the fund receiver if the escrow has funds
+// only the creator
+// can delete the app
+byte "Creator"
+app_global_get
+txn Sender
+==
+
+// check that we are past fund close date
+// global LatestTimestamp
+// byte "FundCloseDate"
+// app_global_get
+// >=
+// &&
+bz failed
+
+// if escrow balance is zero
+// let the app be deleted
+// escrow account must be passed
+// into the call as an argument
+int 0
+int 1
+balance
+==
+
+// if the balance is 0 allow the delete
+bnz finished
+
+// if the escrow is not empty then 
+// there must be need two transactions
+// in a group
+global GroupSize
+int 2
+==
+
+// second tx is an payment
+gtxn 1 TypeEnum
+int 1
+==
+&&
+
+// the second payment transaction should be
+// a close out transaction to receiver
+byte "Receiver"
+app_global_get
+gtxn 1 CloseRemainderTo
+==
+&&
+
+// the amount of the payment transaction 
+// should be 0
+gtxn 1 Amount
+int 0
+==
+&&
+
+// the sender of the payment transaction
+// should be the escrow account
+byte "Escrow"
+app_global_get
+gtxn 1 Sender
+==
+&&
+bz failed
+int 1
+return
+
+not_deletion:
+// check if this is update
+int UpdateApplication
+txn OnCompletion
+==
+
+bz not_update
+// verify that the creator is
+// making the call
+byte "Creator"
+app_global_get
+txn Sender
+==
+
+// the call should pass the escrow 
+// address
+txn NumAppArgs
+int 1
+==
+&&
+bz failed
+
+// store the address in global state
+// this parameter should be addr:
+byte "Escrow"
+txna ApplicationArgs 0
+app_global_put
+int 1
+return
+
+not_update:
+// check for closeout
+int CloseOut
+txn OnCompletion
+==
+bnz close_out
+
+// check if no params are
+// passed in, which should
+// only happen when someone just
+// wants to optin
+// note the code is written
+// to allow opting in and donating with
+// one call
+int 0
+txn NumAppArgs
+==
+bz check_parms
+
+// Verify someone is 
+// not just opting in
+int OptIn
+txn OnCompletion
+==
+bz failed
+int 1
+return
+
+check_parms:
+// donate
+txna ApplicationArgs 0
+byte "donate"
+==
+bnz donate
+
+// reclaim
+txna ApplicationArgs 0
+byte "reclaim" 
+==
+bnz reclaim
+
+// claim
+txna ApplicationArgs 0
+byte "claim" 
+==
+bnz claim
+b failed
+
+
+donate:
+// check dates to verify
+// in valid range
+// Note: Timestamp check is excluded from this example, 
+// To add it, remove comments from following lines.
```

Comment:
Please move the above 4 comments before the branch declaration (above `donate:`)

Reply:
i don't think that is good, because these comments are for `timestamp` not for whole `donate`

Reply:
Then can we move this to top (if there are other places with commented timestamp code)?
If not, can we add a new line before the commented code starts (it looks like a big comment otherwise)?

Reply:
i think we can add a line,.